http/oidc/src/main/java/org/wildfly/security/http/oidc/OidcHttpFacade.java

```diff
@@ -205,9 +201,10 @@ public String getMethod() {
             @Override
             public String getURI() {
                 try {
-                    return URLDecoder.decode(request.getRequestURI().toString(), "UTF-8");
-                } catch (UnsupportedEncodingException e) {
-                    throw log.failedToDecodeRequestUri(e);
+                    URI decode = new URI(null, null, request.getRequestURI().toString(), null);
```

Comment:
@kanha-gupta Just a minor, after you can successfully build the project you will see if you have test failures after this change. 

In the description of this issue, it is suggested to decode individual URL parameters instead of the entire URI. URL parameters can be taken from URI and are made of a key and a value separated by an equals sign (=) and joined by an ampersand (&) . IIUC, you are decoding full URI path here instead of the individual URL parameters.